spring-kafka/src/main/java/org/springframework/kafka/listener/GenericMessageListener.java

```diff
@@ -17,20 +17,15 @@
 package org.springframework.kafka.listener;
 
 /**
- * Top level interface for listener types using container or auto commits.
+ * Top level interface for listener types using container commits or auto commits.
  *
  * @param <T> the type.
  *
  * @author Gary Russell
  * @since 1.1
  *
  */
+@FunctionalInterface
 public interface GenericMessageListener<T> extends KafkaDataListener<T> {
 
-	/**
-	 * Invoked with data from kafka.
-	 * @param data the data to be processed.
-	 */
-	void onMessage(T data);
```

Comment:
Yes, we can remove it now (since it's a major release). This was needed before because we needed a common super-interface for `GenericAcknowledgingMessageListener` and `GenericMessageListener`.